### ISSUE TITLE
Git not detecting changes

### ISSUE THREAD

**Issue by marcbeldata** [rstudio/rstudio#2035]
My system:
    OS Version      :  High Sierra
    RStudio Version : 1.1.414
    R Version       : 3.4.3

I'm trying to use git in a new machine (previously I've never work with git inside RStudio). Git seems properly installed:

<img width="654" alt="terminal1" src="https://user-images.githubusercontent.com/15173027/35265422-55d7744a-0020-11e8-9b92-dd9bc6f93ee0.png">

![rstudio1](https://user-images.githubusercontent.com/15173027/35265531-9fa76c42-0020-11e8-8d8a-77a08dee4e23.png)

I create a new project in RStudio with my public test repo in github (https://github.com/marcbeldata/testRepo) (in local, not github, I'm having the same problem), and created a script (File.R). Nothing is detected in git window:

<img width="1151" alt="rstudio2" src="https://user-images.githubusercontent.com/15173027/35265848-a511532c-0021-11e8-8575-382a4f170658.png">

Update:

It seems it affects existing projects you add git version control . With new ones from scratch (File>New Project>New Directory>New Project, Create a git repository) works fine.

But with existing projects you need to make a first commit from Terminal.

**Comment by ronblum** [comment]
@marcbeldata I've tried reproducing the problem, but when I do so, the file shows up. Can you describe the exact steps you used to reproduce the problem, in the same way you did in the update?

**Comment by marcbeldata** [comment]
1.- In Finder I create a new folder with files (i.e. "NewProjectAgain), before starting the new project.

2.- In RStudio, File>New Project > Existing Directory>Browse, I select the folder ("~/Documents/NewProjectAgain"), Create Project. 

(New) Problem. In Tools>Project Options, it's assigning by default a github repository I didn't enter in this  project (?). 

<img width="1364" alt="problemgit" src="https://user-images.githubusercontent.com/15173027/35378103-dbe155aa-01b1-11e8-9b0f-f361a16a8976.png">

This is the repository of the first post in this thread,  entered by RStudio>File>New Project>Version Control>Git.

3.- I create a new script ("NewScript.R), add some text. Git is not detecting this new file.

<img width="1274" alt="git3" src="https://user-images.githubusercontent.com/15173027/35378608-bb791bac-01b3-11e8-8a2e-152595a3b0e9.png">

**Comment by ronblum** [comment]
Thank you! 
1) I was able to reproduce the original scenario by using an accented "ó" in the directory name—like the original name, "testRepo — Comunicació", and "comó". Changing the name to "como" (no accent) and reopening it works—files show up in the git pane as expected. The problem also occurs of any directory in the path has an accented letter in it; for example, pointing the project at the directory ".../tódo/newrep" causes the same problem.

2) That said, I'm not sure why the folder "NewProjectAgain" is causing a problem. I can't reproduce this. What is the path to the home directory—does it have a non-ASCII character in it, by chance?

3) For the new issue, it looks as if the home directory might've been initialized as a git repository. Can you switch to the home directory on the command line and run `git status`?

**Comment by marcbeldata** [comment]
2.- No. "NewProjectAgain" is in Macintosh HD>Users>Home (marcbelzunces)>Documents.

My system was installed in English (so no accents in folders created by OS), which is the current primary language (Catalan as secondary), while Region is set to "Spain" (unmodified). Locale in RStudio is en_US.UTF-8.

3.- I get this:

<img width="654" alt="terminal" src="https://user-images.githubusercontent.com/15173027/35438651-a21393f6-0297-11e8-984c-f6ac1c8e6ae6.png">

**Comment by ronblum** [comment]
Ah! The home directory is being treated as a repository, with "NewProjectAgain" as a subfolder/project under it, which is why all the home directory files showed up in the git pane. Assuming you don't want the the home directory to be treated as a repository, removing the "~/.git" directory will take care of it.

**Comment by marcbeldata** [comment]
Thanks Ron! It works correctly now!

The quesiton is that in the initial problems I just followed the official instructions from  [https://support.rstudio.com/hc/en-us/articles/200532077?version=1.1.419&mode=desktop](https://support.rstudio.com/hc/en-us/articles/200532077?version=1.1.419&mode=desktop), that is, clean installation of git (Download, click-click-clik, installed) on a workstations with R&RStudio installed 1,5 month ago..., New Project, etc.

**Comment by avecchiato** [comment]
Hi, I've had a similar problem that was not solved by the suggestion above. I've followed all the instruction to create an R project with version control. My issue is that if a R script is first created on Github that it gets tracked by version control. If I create new rscripts directly in R Studio and then save them they are not tracked. Rscript files are obviously not included in .gitignore and I don't understand why they are not tracked! Any idea?

**Comment by Sumidu** [comment]
I have had the same Problem with a Project in a sub-sub-subfolder of a Folder that contains an Umlaut (Ü). Since in a teaching environment in German often has folders with umlauts (e.g. "Übung" is Tutorial" etc.).
As soon as I removed all Umlauts RStudio became aware of changes in git.

**Comment by shmuhammad2004** [comment]
I solved my problem. I removed my folder which was in Dropbox folder. There are many issues that can cause this problem . Read [chapter 14](https://happygitwithr.com/) of Happy Git and R. I found my solution there.

**Comment by shmuhammad2004** [comment]
> Hi, I've had a similar problem that was not solved by the suggestion above. I've followed all the instruction to create an R project with version control. My issue is that if a R script is first created on Github that it gets tracked by version control. If I create new rscripts directly in R Studio and then save them they are not tracked. Rscript files are obviously not included in .gitignore and I don't understand why they are not tracked! Any idea?

I solved my problem. I change location of my folder which was in Dropbox folder. There are many issues that can cause this problem . Read chapter 14 of Happy Git and Github for R Users. I found my solution there.

**Comment by jmcphers** [comment]
These issues should be resolved in the dailies now that we set the encoding correctly when invoking Git. Let us know if you're still seeing them in build > 1.3.610!

**Comment by tormodb** [comment]
This bug reappeared in Rstudio version 1.4.1100 when using the Norwegian letter "å" in the name of a folder.

**Comment by kevinushey** [comment]
@tormodb: can you please file a new issue with a reproducible example of what you're seeing?
